server/src/main/java/org/elasticsearch/index/search/MultiMatchQueryParser.java

```diff
@@ -106,15 +99,216 @@ private Query combineGrouped(List<Query> groupQuery, float tieBreaker) {
                 queries.add(query);
             }
         }
-        return queries;
+        return combineWithDisMax(queries, tieBreaker);
     }
 
-    private List<Query> buildCrossFieldQuery(Map<String, Float> fieldNames,
-                                             Object value, String minimumShouldMatch, float tieBreaker) {
+    private Query combineWithDisMax(List<Query> groupQuery, float tieBreaker) {
+        if (groupQuery.isEmpty()) {
+            return zeroTermsQuery();
+        }
+        if (groupQuery.size() == 1) {
+            return groupQuery.get(0);
+        }
+        return new DisjunctionMaxQuery(groupQuery, tieBreaker);
+    }
+
+    private Query buildCombinedFieldQuery(Map<String, Float> fields, Object value, String minimumShouldMatch) {
+        validateSimilarity(fields);
 
+        List<Query> queries = new ArrayList<>();
+        Map<Analyzer, List<FieldAndBoost>> groups = new HashMap<>();
+        for (Map.Entry<String, Float> entry : fields.entrySet()) {
+            String name = entry.getKey();
+            MappedFieldType fieldType = context.getFieldType(name);
+            if (fieldType == null) {
+                continue;
+            }
+
+            if (fieldType.getTextSearchInfo() != TextSearchInfo.NONE) {
+                float boost = entry.getValue() == null ? 1.0f : entry.getValue();
+                Analyzer actualAnalyzer = getAnalyzer(fieldType, false);
+                if (groups.containsKey(actualAnalyzer) == false) {
+                    groups.put(actualAnalyzer, new ArrayList<>());
+                }
+                groups.get(actualAnalyzer).add(new FieldAndBoost(fieldType, boost));
+            } else {
+                IllegalArgumentException exception = new IllegalArgumentException("Field [" + fieldType.name() +
+                    "] of type [" + fieldType.typeName() + "] does not support match queries");
+                if (lenient) {
+                    queries.add(newLenientFieldQuery(name, exception));
+                } else {
+                    throw exception;
+                }
+            }
+        }
+
+        for (Map.Entry<Analyzer, List<FieldAndBoost>> group : groups.entrySet()) {
+            // Pick an arbitrary field name to pass through the builder.
+            String field = group.getValue().get(0).fieldType.name();
+            MatchQueryBuilder builder = new CombinedFieldsQueryBuilder(group.getKey(), group.getValue(),
+                enablePositionIncrements, autoGenerateSynonymsPhraseQuery);
+            Query query = builder.createBooleanQuery(field, value.toString(), occur);
+            if (query == null) {
+                query = zeroTermsQuery();
+            }
+            query = Queries.maybeApplyMinimumShouldMatch(query, minimumShouldMatch);
+            if (query != null) {
+                queries.add(query);
+            }
+        }
+
+        return combineWithBoolean(queries);
```

Comment:
I am wondering what is the reason we are not using dis_max and tie_breaker for groups  in a `combined_field` similar as we did for groups in `cross_fields`?

Reply:
This is a good question, I forgot to make a note about this. I chose to combine the groups with boolean 'should' for a few reasons. First, boolean is simpler than dismax in that it doesn't require an extra query parameter, and a more common default.

Dismax is also not a clear fit scoring-wise. It's helpful when you want to reward matches where all terms are present in a single group, but we don't necessarily want to boost these matches. For example, there could be different groups because each field was analyzed in different ways (exact, stemmed, etc.) Then we would want to sum the evidence, as we suggest in the [most_fields docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html#type-most-fields). In the [cross_fields docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-multi-match-query.html#cross-field-analysis) we actually claim that groups are combined through a boolean 'should'. So I wonder if that was our intention even for cross_fields.

Reply:
I opened a PR to correct the `cross_fields` docs: #69936